oci/layout/oci_transport.go

```diff
@@ -228,7 +247,55 @@ func (ref ociReference) NewImageDestination(ctx context.Context, sys *types.Syst
 
 // DeleteImage deletes the named image from the registry, if supported.
 func (ref ociReference) DeleteImage(ctx context.Context, sys *types.SystemContext) error {
-	return errors.New("Deleting images not implemented for oci: images")
+	// Get the manifest for the image
+	descriptor, err := ref.getManifestDescriptor()
+	if err != nil {
+		return err
+	}
+
+	manifest, err := ref.getManifest(descriptor)
+	if err != nil {
+		return err
+	}
+
+	// Delete all blobs
+	for _, d := range append(manifest.Layers, manifest.Config, descriptor) {
+		blobPath, err := ref.blobPath(d.Digest, "")
+		if err != nil {
+			return err
+		}
+		err = os.Remove(blobPath)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Update the index
+	index, err := ref.getIndex()
+	if err != nil {
+		return err
+	}
+
+	newManifests := make([]imgspecv1.Descriptor, len(index.Manifests)-1)
+	for i, v := range index.Manifests {
+		if v.Annotations[imgspecv1.AnnotationRefName] != ref.image {
```

Comment:
`ref.image` is not always set (and there is some WIP work to also allow `ociReference` to refer to an image directly using an index.

I guess `ref.getManifest` could also return an index.

Reply:
You mean that ref.image can be "", as described in here: https://github.com/containers/image/blob/main/oci/layout/oci_transport.go#L64 ?

When you say index, you mean that someone could ask deletion of the Nth image in the index.json instead of giving the name? (in that case, that would mean that ref.image being "" is the same as index being 0 and only one image)? I don't understand an index to what `getManifest` would return, unless you meant `getManifestDescriptor` which would return the index at which it found the image in the `index.json`?

Reply:
> You mean that ref.image can be ""

Yes.

> When you say index, you mean that someone could ask deletion of the Nth image in the index.json instead of giving the name? 

Exactly. That feature already exists for `docker-archive:`, and there are WIP PRs around implementing the same thing for the two OCI transports.

> I don't understand an index to what `getManifest` would return, unless you meant `getManifestDescriptor`

Yes. My mistake.

Reply:
All right, makes sense. I'll leave that for now not to clutter the issue worked, but I'm happy to add this later on when you feel the code is good enough.